### ISSUE TITLE
Sending data question

### ISSUE THREAD

**Issue by ZCKevin-TW** [NTU-SP/SP-Hw1-release#20]
Hi, I did some self test with the following
`echo "902001\nExit" | nc localhost [some_port]`
and it turns out the data `902001\nExit` is sent altogether.
In this case, when reading from `read_server`, it might happens that `902001\nExit\n` is reaceived.
But using the current method to process it, I might cut the input to `902001` and totally ignore `Exit`.
Thus the connection is not closed by the server.

Do we need to handle such case?
Or we can assume that when reading, the data followed by one `\n` will never be sent together.

**Comment by hi-i-m-GTooth** [comment]
In this case, you created an invalid input, `902001\nExit\n`, which is not in our invalid-input guarantee (refer to spec sec 5.4).
Hence, you don't need to handle this type of input for this assignment.
But you could consider this situation in your future work.

**Comment by ZCKevin-TW** [comment]
Thanks for helping, but I still have some questions.
Why is this invalid ?
`902001\nExit\n` is equivalent with 
`902001` <hit enter>
`Exit` <hit enter>
and due to telnet default mode(line mode), it sends data when a new line(hit enter) event occur.
but it's possible that telnet use other mode to send data( all message in a pack ) like above.

More over, it's also possible that 
1. client A use telnet sends `902001` to our server
2. our server is processing other client, so busy that it hasn't handle client A yet.
3. client A sends `Exit`.
4. now both `902001` and `Exit` are present in the stream, so when we use `read` to read it, it comes as `902001\nExit\n`. 
5. when it's read together, our server might need a lot more code to handle such situation

The question is whether we need to handle this?

**Comment by ZCKevin-TW** [comment]
Or is it guaranteed that user "always" send "Exit" only after the user received "(Type Exit  to leave...)" message ?

**Comment by imingasco** [comment]
I think the spec shows that you should not send `902001\nExit\n' clearly.
![Uploading 截圖 2022-10-07 下午5.21.40.png…]()

**Comment by imingasco** [comment]
> 902001\nExit\n is equivalent with
> 902001 (hit enter)
> Exit (hit enter)

This is not true. In telnet or other interactive client, data in buffer will be sent whenever you hit enter. In your method, server will receive ` 902001\nExit\n` in one `handle_read`, which is different from the specified behavior.
